### PR TITLE
DTSPO-8329: Upgrade Traefik version in Demo Environment

### PR DESCRIPTION
**Pull request by reespozzi**
W.I.P PR to upgrade Traefik in demo

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

**Comment by reespozzi** [comment]
Still to look at: external-dns-public, aad-pod-mi, traefik-auth & oauth2-proxy, patches and config to get traefik-private and traefik working.

**Comment by reespozzi** [comment]
Generated and added all secrets to demo KV according to [docs](https://oauth2-proxy.github.io/oauth2-proxy/docs/configuration/oauth_provider/#azure-auth-provider)

**Comment by reespozzi** [comment]
TODO: DNS record for oauth needs looking at, redirects & testing certain app on new cluster, vault reader permissions from cluster

**Comment by reespozzi** [comment]
TODO: Secret provider for traefik2 in demo

**Comment by reespozzi** [comment]
TODO: Key vault permissions https://github.com/hmcts/aks-cft-deploy/pull/246

File: apps/admin/demo/01/kustomization.yaml [hunk 1]
```diff
@@ -1,4 +1,24 @@
...
+
+patchesStrategicMerge:
+  - ../../external-dns-2/demo/01-external-dns.yaml
+  - ../../external-dns-2/demo/external-dns-private.yaml
+  - ../../external-dns-2/demo/01-external-dns-private.yaml
+  - ../../traefik2/demo/01.yaml
+  - ../../traefik2/demo/traefik-private.yaml
+  - ../../traefik2-no-proxy/demo/01.yaml
+  - ../../traefik2-auth/demo/01.yaml
+  - ../../oauth2-proxy/demo/01-oauth2-proxy.yaml
```
[the remark this answers is not in the document]
Comment (reespozzi) [reply apps/admin/demo/01/kustomization.yaml:23]: https://github.com/hmcts/cnp-flux-config/pull/18289/files

File: apps/admin/aad-pod-id/mi/demo/azure-identity-patch.yaml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+apiVersion: "aadpodidentity.k8s.io/v1"
```
[the remark this answers is not in the document]
Comment (reespozzi) [reply apps/admin/aad-pod-id/mi/demo/azure-identity-patch.yaml:1]: https://github.com/hmcts/cnp-flux-config/pull/17695/commits/9889bd42b21839f00a26c659b4e89dee5659c1b1

File: apps/admin/external-dns-2/demo/01-external-dns.yaml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+apiVersion: helm.toolkit.fluxcd.io/v2beta1
+kind: HelmRelease
+metadata:
+  name: external-dns
+  namespace: admin
+spec:
+  releaseName: external-dns
+  values:
+    logLevel: debug
+    txtOwnerId: cft-demo-aks-active
```
[the remark this answers is not in the document]
Comment (reespozzi) [reply apps/admin/external-dns-2/demo/01-external-dns.yaml]: https://github.com/hmcts/cnp-flux-config/pull/17695/commits/7ab113587948daa6835d75da23e39ab1e0880f9a

File: apps/admin/demo/01/kustomization.yaml [hunk 4]
```diff
@@ -1,4 +1,24 @@
...
+  - ../base
+  - ../../secrets-csi-driver-crds
+  - ../../secrets-csi-driver
+  - ../../traefik2
+  - ../../traefik2-no-proxy
+  - ../../traefik2-auth
+  - ../../external-dns-2/
+  - ../../aad-pod-id/mi
+  - ../../oauth2-proxy
+  - ../../external-dns-2/demo/external-dns.yaml
```
[the remark this answers is not in the document]
Comment (reespozzi) [reply apps/admin/demo/01/kustomization.yaml]: Attempted this in https://github.com/hmcts/cnp-flux-config/pull/17695/commits/583d113d409f6560ed8ae14bfe0bf9e1448400d7 @adusumillipraveen

File: apps/admin/external-dns-2/external-dns-public/external-dns-public.yaml [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+  namespace: admin
+  annotations:
+    flux.weave.works/automated: "false"
+spec:
+  releaseName: external-dns
+  values:
+    provider: azure
+    annotationFilter: "kubernetes.io/ingress.class=traefik"
+    azure:
+      resourceGroup: reformMgmtRG
```
Comment (reespozzi) [inline apps/admin/external-dns-2/external-dns-public/external-dns-public.yaml:14]: check if this RG is correct for CFT public dns

**Comment by hannah38** [comment]
Extra external-dns release from https://github.com/hmcts/cnp-flux-config/blob/DTSPO-8329-Add-demo-traefikv2/apps/admin/external-dns-2/kustomization.yaml#L6 - should just need removed now it's coming from private/public

**Comment by reespozzi** [comment]
> Member

[339fa56](https://github.com/hmcts/cnp-flux-config/pull/17695/commits/339fa5632a3c8dd5a605cd453bd42de348278197)

**Comment by reespozzi** [comment]
> Extra external-dns release from https://github.com/hmcts/cnp-flux-config/blob/DTSPO-8329-Add-demo-traefikv2/apps/admin/external-dns-2/kustomization.yaml#L6 - should just need removed now it's coming from private/public

[339fa56](https://github.com/hmcts/cnp-flux-config/pull/17695/commits/339fa5632a3c8dd5a605cd453bd42de348278197)

**Comment by reespozzi** [comment]
Tests here passing but kustomize build failing locally in demo/01 - refactored traefik kustomization code in https://github.com/hmcts/cnp-flux-config/pull/18414

**Pull request merged**